### ISSUE TITLE
LARMOR, ZOOM or SURF: Knauer HPLC 1050 standardise pump interface to match JASCO 4180

### ISSUE THREAD

**Issue by aaron-long** [ISISComputingGroup/IBEX#4004]
As a scientist I would like to be able to have parity of the interface design between devices that have identical functionality.

The Knauer 1050 is a subpar implementation of an HPLC device in comparison to the JASCO 4180. The device does not have the matching functionality to the VI it superseded (See: #3881). It's interface is also objectively worse also.

The Knauer 1050 OPI should be updated to match the standard that the JASCO 4180 has set.

Requires #3881

**Comment by kjwoodsISIS** [comment]
As a general rule, when implementing an IOC & OPI for a device that already has a LabVIEW VI, we should be seeking to provide parity of functionality (unless there is a good reason not to do so).  The OPI does not have to look identical to the LabVIEW VI but, at a minimum, it should provide access to the same device functionality.
If there is a good reason to provide only a reduced level of functionality, then please make sure you explain this to the instrument scientist.

**Comment by aaron-long** [comment]
Of course, however, in the case of the Kanuer it was needed urgently for the SURF experiment. As such the work to provide timed/volume pump  was seen as an additional extra and not included. This was confirmed with the scientist before the IOC was provided.

The JASCO (in essence) was the second iteration of an IBEX HPLC, and as such the lessons learnt made it better. Also the GUI design was made to be much clearer and the scientists commented:

•	Clear separation of what is read back from the device (current state) from what is desired (set point)
•	Obvious how to specify a new set point and how to action it

The Knauer need to be better; I made this ticket because I will be doing it as a Friday ticket.

**Comment by aaron-long** [comment]
- IBEX_GUI: https://github.com/ISISComputingGroup/ibex_gui/pull/856

**Comment by Alistair-McGann-Tessella** [comment]
Ticket #3881 must be merged before reviewing this ticket

**Comment by aaron-long** [comment]
I should have really completed #3881 and #4004 as a single ticket, as they both rely on each other. I added the requires flag but I realise now that's not the same thing.

**Comment by aaron-long** [comment]
Separated the Pump control and Device status for clarity as requested.

**Comment by aaron-long** [comment]
Please see Ticket [4143](https://github.com/ISISComputingGroup/IBEX/issues/4143) - The work carried out in this ticket was completed in this new ticket.

**Comment by Alistair-McGann-Tessella** [comment]
Can this ticket be closed as now a duplicate?

**Comment by aaron-long** [comment]
No, this should be considered as part of #4143. I made the mistake of making two tickets when each change relied on the other tickets changes.
